packages/web-frontend/src/containers/OverlayDiv/components/ProjectPage/ProjectCardList.js

```diff
@@ -5,26 +5,51 @@
 
 import React from 'react';
 import { ProjectCard } from './ProjectCard';
+import {
+  ALLOWED_PROJECT_ACCESS_TYPE,
+  DENIED_PROJECT_ACCESS_TYPE,
+  PENDING_PROJECT_ACCESS_TYPE,
+} from '../../../../constants';
 
 const EXPLORE_CODE = 'explore';
 
-export const ProjectCardList = ({ projects, accessType, action, actionText }) => {
-  const hasPendingType = accessType === 'pending';
-  const hasAccessType = hasPendingType ? false : accessType;
-  return projects
-    .filter(
-      ({ code, hasAccess, hasPendingAccess = false }) =>
-        code !== EXPLORE_CODE && hasAccess === hasAccessType && hasPendingAccess === hasPendingType,
-    )
-    .sort((a, b) => a.sortOrder - b.sortOrder)
-    .map(project => (
-      <ProjectCard
-        key={project.name}
-        projectAction={() => action(project)}
-        actionText={actionText}
-        accessType={hasAccessType}
-        hasAccessPending={hasPendingType}
-        {...project}
-      />
-    ));
+export const ProjectCardList = ({ projects, actions, isUserLoggedIn }) => {
+  const actionText = {
+    [PENDING_PROJECT_ACCESS_TYPE]: 'Approval in progress',
+    [ALLOWED_PROJECT_ACCESS_TYPE]: 'View project',
+    [DENIED_PROJECT_ACCESS_TYPE]: isUserLoggedIn ? 'Request access' : 'Log in',
+  };
+  const sortedProjects = [
+    ALLOWED_PROJECT_ACCESS_TYPE,
+    PENDING_PROJECT_ACCESS_TYPE,
+    DENIED_PROJECT_ACCESS_TYPE,
+  ].reduce((result, accessType) => {
+    const action = actions[accessType];
```

Comment:
One minor maintenance concern is that the actions have to have these 3 keys otherwise it will break. Maybe we could add prop types for the actions and/or hardcode the actions into 3 props to reflect the project types.